bart/sched/pelt.py

```diff
@@ -0,0 +1,347 @@
+        self.period_us = Task.pelt_sample_us * period_ps
+        self.run_us    = Task.pelt_sample_us * run_ps
+
+        self.pelt_sample_us = pelt_sample_us
+        self.pelt_max = pelt_max
+
+        self.idle_us   = self.period_us - self.run_us
+        self.duty_cycle_pct = int(100. * run_ps / period_ps)
+        self.util_avg =  int(float(Task.pelt_max) * self.duty_cycle_pct / 100)
+
+    def running(self, time_ms):
+        """
+        Return wheter the task is running in the specified time instant.
+
+        The current implementation provide support only for periodic tasks
+        which have the following (abstract) execution model:
+
+            +----+    +----+      Task is RUNNING
+            |    |    |    |
+                 +----+    +----  Task is SLEEPING
+
+           -+----+----+----+-------------------------- Time --->
+            1    2    3    4      PELT Samples
+           ^    ^    ^    ^       Task running sampling time
+           R    S    R    S       Task Status reported by this function
+
+
+         Since we sample exactly at PELT samples intervals, the exact time when
+         a task starts to run we consider it as if if was running for the
+         entire previous PELT interval. To fix this, by exploting the knowledge
+         that a task is always configured using an integer number of run/speep
+         PELT samples, we can look at the 1us before time wrt the specified
+         sample time.
+        """
+        # The 1us "before time" is always represeting the exact status of the
+        # task in that period. This is why we remove 1 in the following
+        # computation:
+        period_delta_us = (float(time_ms) * 1e3 - 1) % self.period_us
+        if period_delta_us <= self.run_us:
+            return True
+        return False
+
+    def __str__(self):
+        return "Task(start: {:.3f} [ms], period: {:.3f} [ms], run: {:.3f} [ms], "\
+               "duty_cycle: {:.3f} [%], util_avg: {:d})"\
+                .format(self.start_us/1e3, self.period_us/1e3,
+                        self.run_us/1e3, self.duty_cycle_pct,
+                        self.util_avg)
+
+class PeriodicTask(Task):
+    """
+    A periodic task which is specified by period and duty-cycle.
+
+    :see also: Task
+    """
+    def __init__(self, period_ps, duty_cycle_pct, start_ps=0):
+        run_ps = period_ps * float(duty_cycle_pct) / 100
+        super(PeriodicTask, self).__init__(period_ps, run_ps, start_ps)
+
+class AndroidTask(Task):
+    """
+    An "android" task which is specified by run-time over a (default) period
+    of 16ms.
+
+    :see also: Task
+    """
+    def __init__(self, run_ps, start_ps=0, period_ps=16):
+        super(AndroidTask, self).__init__(period_ps, run_ps, start_ps)
+
+
+################################################################################
+# PELT Model
+################################################################################
+
+PELT_Sim = _namedtuple('PELT_Sim', [
```

Comment:
We can just name these variables in a camel cased way.

`PeltStats`

or `PELTStats`

whichever you prefer.